### ISSUE TITLE
2 wallpapers overlaps when restore + random with swww

### ISSUE THREAD

**Issue by dennkaii** [anufrievroman/waypaper#9]
![image](https://github.com/anufrievroman/waypaper/assets/70287696/a8c97ff5-a174-418e-9a12-e41132e48d39)

is happening since my last update yesterday i think

i have waypaper --restore --random --backend sww

**Comment by anufrievroman** [comment]
Hmm, these are two different wallpapers on the screenshot that are overlapped/combined? And are they staying this way permanently? 

P.s. I feel it might be `swww` bug but I'll try to reproduce

**Comment by dennkaii** [comment]
> Hmm, these are two different wallpapers on the screenshot that are overlapped/combined? And are they staying this way permanently?
The wallpaper is being restored the thing is when the --random happens they overlap instead of changing.
They stay like that until i change them via waypaper

**Comment by anufrievroman** [comment]
Interesting. I wasn't really able to reproduce it. So, a few more questions:
1) Do you have more than one monitor?
2) Does this happen even if you remove `--random`?
3) May there be some other wallpaper demon that starts with your system, like `hyprpaper` or `swaybg`?
4) If all previous are "no", then could you show your `.config/waypaper/config.ini` ?
Thank you.

**Comment by dennkaii** [comment]
> * Do you have more than one monitor?
No 
> * Does this happen even if you remove `--random`?
no
> * May there be some other wallpaper demon that starts with your system, like `hyprpaper` or `swaybg`?
i dont think so i installed swaybg and the program started working when i see the logs swaybg is not being launched

> 4\. If all previous are "no", then could you show your `.config/waypaper/config.ini` ?
![image](https://github.com/anufrievroman/waypaper/assets/70287696/bb05d6ba-b182-4c40-be71-f7c314b45f20)

**Comment by anufrievroman** [comment]
Hmm, everything seems very normal. I wonder where it takes two images because wallpaper setting is supposed to happen only once with one image. Sorry, I'll need to ask a few more questions:
1) Every time it is two random images that are overlapped or one of them is always the same (or maybe one is from the previous run?)
2) When you just run `waypaper --restore --random --backend swww` in a terminal, what is the output? 
3) If you have only animated gif wallpapers, I noticed that `swww` sometimes treats them strangely and takes a while to start playing them, I don't know if it is related or happens with all kinds of wallpapers to you.

**Comment by dennkaii** [comment]
> 2. When you just run `waypaper --restore --random --backend swww` in a terminal, what is the output?

![image](https://github.com/anufrievroman/waypaper/assets/70287696/74c24c67-3f07-42ef-92b4-bd2c20c7ed61)


> 1. Every time it is two random images that are overlapped or one of them is always the same (or maybe one is from the previous run?)

always the previous one with the new one on top




> 3\. If you have only animated gif wallpapers, I noticed that `swww` sometimes treats them strangely and takes a while to start playing them, I don't know if it is related or happens with all kinds of wallpapers to you.

they are all gifs, i noticed this too but strangely both are running at the same time

**Comment by anufrievroman** [comment]
Thank you for checking. I knew it will be nix :D
Again, everything seems to be fine, those errors in the output are normal and the output shows that indeed the command to set the wallpaper runs successfully and only once. 

Okay, then just to make sure, is it working normally if you simply run:
```
swww img some_new_image.gif
```
where `some_new_image.gif` is the path to any image different from the current wallpaper.

**Comment by anufrievroman** [comment]
And also, what happens if you run `swww query`? Does it show only one wallpaper, and if yes, which one?

**Comment by dennkaii** [comment]
> And also, what happens if you run `swww query`? Does it show only one wallpaper, and if yes, which one?

![image](https://github.com/anufrievroman/waypaper/assets/70287696/84e049d7-0161-4b87-8ab9-c3719dec12b9)

Shows the new wallpaper

**Comment by anufrievroman** [comment]
Okay, good. But how about the first question? I mean, if you simply run
```
swww img some_new_image.gif
```
where some_new_image.gif is the path to any image different from the current wallpaper, does it work?

**Comment by dennkaii** [comment]
> Okay, good. But how about the first question? I mean, if you simply run
> 
> ```
> swww img some_new_image.gif
> ```
> 
> where some_new_image.gif is the path to any image different from the current wallpaper, does it work?

yea it works

**Comment by anufrievroman** [comment]
I see, so it probably happens due to `waypaper` itself. It seems that it is a pretty unique issue, I tried to reproduce it on several different machines but didn't see anything like that. I wonder it could be in your unique hardware or something to do with Nix. By chance, did you try it in other systems or the same Nix system but on adifferent machine?

**Comment by dennkaii** [comment]
> By chance, did you try it in other systems or the same Nix system but on adifferent machine?

I haven't nor have another linux machine to test it sadly

**Comment by anufrievroman** [comment]
Okay. To be honest, I am out of ideas why that could happen. I've been planing to install Nix anyway, so I'll try to see if it is OS specific.

**Comment by dennkaii** [comment]
hmmm, actually now that i think about it could it be that both backends swww and swaybg are trying to set wallpapers?

**Comment by anufrievroman** [comment]
It can only happen if somewhere in your system you start swaybg deamon. In the waypaper, there's a clear if-else condition that runs only currently selected backend. And like we saw previously, it reports that it sends command for swww, so who is setting/maintaining the previous wallpaper is a mystery :) maybe you can try to check if swaybg is running among processes, or even try to delete swaybg from the system? In any case, swaybg does not support gifs.

You can also see `changer.py` file for yourself, it's a fairly simple code that just sends standard commands, I can't see how it can run twice to make two wallpapers. Maybe you could try to remove all mentions of all wallpaper programs from your system startup and then make sure that after reboot there is no wallpaper showing up?

**Comment by dennkaii** [comment]
[https://streamable.com/pqo1kx](url)


> Maybe you could try to remove all mentions of all wallpaper programs from your system startup and then make sure that after reboot there is no wallpaper showing up?

the video is on streamble bc is too big apparently but after trying out the issue happens 2/3 but is weird bc on the first try it fixed after killing the sww-daemon but in the second one it persist.

**Comment by anufrievroman** [comment]
Oh, I am so sorry, I couldn't watch it before expiration as I was on vacation, so if you still have it, please drop it again.

**Comment by anufrievroman** [comment]
Hi, it's been a while with this issue, I could never really reproduce it, nor others seem to experience it. I'll close this for no activity. If the problem persists, feel free to open the new issue. Thank you for reporting.
